### PR TITLE
fix: add missing dep files to manifest

### LINKED ISSUE #2484: bug/PyPi source distribution missing requirements files
**Describe the bug**

The [source distribution](https://pypi.org/project/unstructured/#files) available on the project's [PyPi page](https://pypi.org/project/unstructured/) contains only two requirements files - `base.in` and `huggingface.in` - and is missing [the rest](https://github.com/Unstructured-IO/unstructured/tree/main/requirements).

**To Reproduce**

```bash
wget https://files.pythonhosted.org/packages/source/u/unstructured/unstructured-0.12.3.tar.gz
tar -zxvf unstructured-0.12.3.tar.gz
ls unstructured-0.12.3/requirements/
# base.in  huggingface.in
pip install ./unstructured-0.12.3
# fails with FileNotFoundError (requirements/extra-csv.in)
```

**Expected behavior**

The source distribution should include the necessary metadata files for setup.py to run.

**Screenshots**

Output of `pip install`:

```text
$ pip install ./unstructured-0.12.3
Looking in indexes: https://pypi.org
Processing ./unstructured-0.12.3
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      Traceback (most recent call last):
        File "[venv]/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
          main()
        File "[venv]/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "[venv]/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/private/var/folders/ps/l7xb9fxn7gb7v9pg6f0gxxlrv6cznd/T/pip-build-env-2w4042p9/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/private/var/folders/ps/l7xb9fxn7gb7v9pg6f0gxxlrv6cznd/T/pip-build-env-2w4042p9/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/ps/l7xb9fxn7gb7v9pg6f0gxxlrv6cznd/T/pip-build-env-2w4042p9/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 480, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "/private/var/folders/ps/l7xb9fxn7gb7v9pg6f0gxxlrv6cznd/T/pip-build-env-2w4042p9/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 42, in <module>
        File "<string>", line 34, in load_requirements
      FileNotFoundError: [Errno 2] No such file or directory: 'requirements/extra-csv.in'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Environment Info**

```text
OS version:  macOS-14.3-x86_64-i386-64bit
Python version:  3.10.9
unstructured is not installed
unstructured-inference is not installed
pytesseract is not installed
Torch is not installed
Detectron2 is not installed
PaddleOCR is not installed
Libmagic version: file-5.41
magic file from /usr/share/file/magic
LibreOffice is not installed
```

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

Closes #2484. Adds missing dependency files to `MANIFEST.in` so they are included in the Python distribution. Also updates the manifest to look for ingest dependencies in the `requirements/ingest` subdirectory.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,23 +1,59 @@
...
+include requirements/ingest/chroma.in
+include requirements/ingest/confluence.in
+include requirements/ingest/databricks-volumes.in
+include requirements/ingest/delta-table.in
+include requirements/ingest/discord.in
+include requirements/ingest/dropbox.in
+include requirements/ingest/elasticsearch.in
+include requirements/ingest/embed-aws-bedrock.in
+include requirements/ingest/embed-huggingface.in
+include requirements/ingest/embed-huggingface.txt
```
Comment (potter-potter) [inline MANIFEST.in:34]: Should this .txt be here?

**Comment by potter-potter** [comment]
Are there downsides to including all these? Will it balloon the release? I'm guessing not since the release is only 1.6 megs right now.
https://pypi.org/project/unstructured/#files

**Comment by MthwRobinson** [comment]
Built the distribution locally, doesn't look like it materially affects the size. Still 1.6M.

```
4112 -rw-r--r--  1 mrobinson  staff   1.6M Feb  6 15:40 unstructured-0.12.4.dev6.tar.gz
```

File: MANIFEST.in (hunk 1 above)
Comment (MthwRobinson) [reply MANIFEST.in]: Donezo! Dropped a few other `.txt` files too

**Review by potter-potter — APPROVED** [review]
Looks good. Not sure if it needs a CHANGELOG and __version__ bump...

File: MANIFEST.in [hunk 2]
```diff
@@ -1,23 +1,56 @@
+# Base requirements and contstraints
```
Comment (qued) [inline MANIFEST.in:1]:
```suggestion
# Base requirements and constraints
```

**Review by qued — APPROVED** [review]
LGTM other than typo

**Comment by MthwRobinson** [comment]
Updated the changelog and bumped the version!

**Comment by potter-potter** [comment]
I'm working on a fix for the OpenSearch CI failure. Not sure why this happens now. But we think it's a port conflict.

**Pull request merged**